File: client/src/application/lessons/dto/semester_lesson_plan.dto.js
class SemesterLessonPlanDTO {

    constructor(data = {}) {

        this.semesterLessonPlanId =
            data.semester_lesson_plan_id;

        this.universityId =
            data.university_id;

        this.academicYearId =
            data.academic_year_id;

        this.academicYearName =
            data.academic_year_name;

        this.semesterId =
            data.semester_id;

        this.semesterName =
            data.semester_name;

        this.semesterNumber =
            data.semester_number;

        this.totalLessons =
            data.total_lessons;
    }
}

module.exports = SemesterLessonPlanDTO;

File: server/apps/lessons/application/dto/semester_lesson_plan_dto.py
from dataclasses import dataclass


@dataclass
class SemesterLessonPlanDTO:

    semester_lesson_plan_id: str

    university_id: str

    academic_year_id: str

    academic_year_name: str

    semester_id: str

    semester_name: str

    semester_number: str

    total_lessons: int